x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticatorChain.java

```diff
@@ -64,7 +65,13 @@ class AuthenticatorChain {
         this.isAnonymousUserEnabled = AnonymousUser.isAnonymousEnabled(settings);
         this.authenticationSerializer = authenticationSerializer;
         this.realmsAuthenticator = realmsAuthenticator;
-        this.allAuthenticators = List.of(serviceAccountAuthenticator, oAuth2TokenAuthenticator, apiKeyAuthenticator, realmsAuthenticator);
+        this.allAuthenticators = List.of(
+            serviceAccountAuthenticator,
+            oAuth2TokenAuthenticator,
+            pluggableApiKeyAuthenticator,
```

Comment:
Does it even make sense to add authenticator to the chain if it's a noop?
We do add others when they are not enabled (e.g OAuth2 token and API key authenticator), but I'm wondering if we should change that (not necessarily in this PR)?